### PR TITLE
.NET 6 Preview 6 API updates

### PR DESCRIPTION
**Pull request by gewarren**
Build.Url: https: //apidrop.visualstudio.com/Content%20CI/_build/results?buildId=231897&view=results
source_repo.branch: net6-preview6
source_repo.url: https://apidrop.visualstudio.com/_git/binaries

### CODE REVIEW

**Review by kouvel — COMMENTED** [review]
System.Threading LGTM

File: xml/System.Runtime.Intrinsics.X86/AvxVnni.xml [hunk 1]
```diff
@@ -0,0 +1,285 @@
+<Type Name="AvxVnni" FullName="System.Runtime.Intrinsics.X86.AvxVnni">
```
Comment (tannergooding) [inline xml/System.Runtime.Intrinsics.X86/AvxVnni.xml:1]: The types under here qualify for: https://github.com/dotnet/dotnet-api-docs/issues/6861

If its possible, we should ensure the `RequiresPreviewFeatures` attribute is visible and that the page comes with some sort of disclaimer that the APIs are preview and are subject to change (including being removed) between now and the next release (up until the attribute is removed).

**Review by tannergooding — APPROVED** [review]
APIs covered by System.Buffers, System.Numerics, System.Runtime, System.Runtime.Intrinsics and System.Text.RegularExpressions LGTM.

I left a comment about the RequiresPreviewFeatures APIs, which we'll likely have more of for Preview 7.

CC. @pgovind for a second pair of eyes

**Review by jkoritzinsky — COMMENTED** [review]
InteropServices LGTM

**Pull request merged**